src/Preload/DllConflictResolver.cs

```diff
@@ -0,0 +1,22 @@
+using System.IO;
+using System.Reflection;
+using BepInEx.Preloader.Core.Patching;
+
+namespace DiscoAPI.Preload;
+
+[PatcherPluginInfo("DiscoAPI.Preload", "DiscoAPI.Preload", "0.0.1")]
+public class DllConflictResolverPatch : BasePatcher
+{
+    private const string ProvidedNewtonsoftPath = "BepInEx/plugins/DiscoAPI/Newtonsoft.Json.dll";
```

Comment:
this is obviously accurate 90+% of the time, so this is more a note for the future than a nit but
there is a `BepInEx.Core.Paths` class for this kind of thing. unfortunately from the way the runtime is set up, the `Plugins` property is not guaranteed to be accurate in a preloader, but `BepInExRootPath` should be accurate. maybe update to run through a list of paths if this at some point breaks.